### PR TITLE
fix: improve performance of agents endpoint for k8s [DET-4073]

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
Previously when we accessed the agents endpoint (WebUI, det slot list, det agent list)
while running on k8s, we would query the k8s API server for the current list of nodes
which was a blocking call. We now keep a running list of current nodes, so there is no
need to make blocking calls to the k8 API server while building the response for the 
agents endpoint.

## Test Plan
Tested manually on a k8 cluster.

### CODE REVIEW

File: master/internal/kubernetes/nodes.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		ctx.Log().Errorf("unexpected message %T", msg)
+		return actor.ErrUnexpectedMessage(ctx)
+	}
+
+	return nil
+}
+
+func (n *nodeInformer) startNodeInformer(ctx *actor.Context) error {
+	nodeInformer := n.informer.Core().V1().Nodes().Informer()
+	nodeInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
```
Comment (eecsliu) [inline master/internal/kubernetes/nodes.go:64]: Question: If a resync period is not specified, does the event handler only update when something changes?

File: master/internal/kubernetes/nodes.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			ctx.Tell(n.podsHandler, nodeStatusUpdate{updatedNode: node})
+		},
+		DeleteFunc: func(obj interface{}) {
+			node := obj.(*k8sV1.Node)
+			ctx.Log().Debugf("node stopped %s", node.Name)
+			ctx.Tell(n.podsHandler, nodeStatusUpdate{deletedNode: node})
+		},
+	})
+
+	ctx.Log().Debug("starting node informer,")
```
Comment (eecsliu) [inline master/internal/kubernetes/nodes.go:82]: nonblocking: is the comma supposed to be there?

**Review by eecsliu — APPROVED** [review]
An optimization worthy of shipment!

File: master/internal/kubernetes/nodes.go (hunk 1 above)
Comment (aaron276h) [reply master/internal/kubernetes/nodes.go:64]: Yeah exactly, there actually is no good reason for us to be doing the resync i  am going to get rid of that.

File: master/internal/kubernetes/nodes.go (hunk 2 above)
Comment (aaron276h) [reply master/internal/kubernetes/nodes.go]: good catch, sloppy fingers on my part.

**Pull request merged**